### PR TITLE
Replace Vector with BitSet in optimization passes

### PR DESCRIPTION
**Pull request by LilithHafner**
I was unable to determine why this sorting operation is needed. I tracked it down to https://github.com/JuliaLang/julia/commit/8af40ea4e7127c1f2736876e42d71f196fad0392 so perhaps Keno (waiting for CI to pass to ping) knows if there is a reason to keep this?

Closes #46507

### CODE REVIEW

File: base/compiler/ssair/passes.jl [hunk 1]
```diff
@@ -1231,7 +1231,7 @@ function sroa_mutables!(ir::IRCode, defuses::IdDict{Int, Tuple{SPCSet, SSADefUse
             else
                 phiblocks = iterated_dominance_frontier(ir.cfg, ldu, get(lazydomtree))
             end
-            allblocks = sort!(vcat(phiblocks, ldu.def_bbs); alg=QuickSort)
+            allblocks = vcat(phiblocks, ldu.def_bbs)
```
Comment (vtjnash) [inline base/compiler/ssair/passes.jl:1234]: It looks like the only query we need is `in`, so it should perhaps be a (Bit)Set, which probably wasn't available at the time this code was initially written
```suggestion
            allblocks = union!(Set(phiblocks), ldu.def_bbs)
```

Comment (LilithHafner) [reply base/compiler/ssair/passes.jl]: Seems reasonable; we'll have to see if it is practically an improvement to use a Set b.c. Vectors are better than Sets for `in` on very small collections.

**Comment by vtjnash** [comment]
@nanosoldier `runbenchmarks("inference", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/46587#issuecomment-1235678098) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/1644480_vs_4690323/report.md).

**Comment by LilithHafner** [comment]
While the performance improvements are negligible, it seems plausible that the previous approach could result in quadratic runtime in certain cases due to https://github.com/JuliaLang/julia/blob/e1139e7209bf09736b07b712549a87c919143012/base/compiler/ssair/passes.jl#L78-L80

I think this is good and the whitespace comes from master, not this PR.

**Comment by LilithHafner** [comment]
Bump (c.f. https://github.com/JuliaLang/julia/pull/46679#issuecomment-1266400035). There doesn't seem to be much practical performance impact of this PR, but it is a theoretical improvement and more importantly, it reduces Core's dependency on Sort, which is needed for https://github.com/JuliaLang/julia/pull/46679.

**Comment by LilithHafner** [comment]
Rerunning benchmarks after rebase
@nanosoldier `runbenchmarks("inference", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/46587#issuecomment-1266556334) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/3146553_vs_c609e9a/report.md).

**Comment by oscardssmith** [comment]
I'm merging this today sans objections.

**Pull request merged**